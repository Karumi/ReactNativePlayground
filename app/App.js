import React from "react";
import App from "./src/App";
import Sentry from "sentry-expo";
import Expo from "expo";

class ReactNativePlaygroundApp extends React.Component {
    constructor(props) {
        super(props)
        this.state = { loading: true };
    }

    async componentWillMount() {
        this.initializeSentry();
        await this.initializeNativeBaseFonts();
        this.setState({ loading: false });
    }

    render() {
        if (this.state.loading) {
            return <Expo.AppLoading />;
        } else {
            return (<App />);
        }

    }

    initializeSentry() {
        Sentry.config('https://b2ba5e1a6cbf4ff79a89291db6ecc3fb@sentry.io/1298369').install();
    }

    initializeNativeBaseFonts() {
        return Expo.Font.loadAsync({
            'Roboto': require('native-base/Fonts/Roboto.ttf'),
            'Roboto_medium': require('native-base/Fonts/Roboto_medium.ttf'),
            'Ionicons': require('@expo/vector-icons/fonts/Ionicons.ttf'),
        });
    }
}

export default ReactNativePlaygroundApp;