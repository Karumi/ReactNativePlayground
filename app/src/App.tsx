import { createStackNavigator } from "react-navigation";
import CameraScreen from "./camera-screen/CameraScreen";
import GridScreen from "./grid-screen/GridScreen";
import ImagePickerScreen from "./image-picker/ImagePickerScreen";
import ListScreen from "./list-screen/ListScreen";
import MainScreen from "./main-screen/MainScreen";
import MapScreen from "./map-screen/MapScreen";
import NavigationScreen from "./navigation-screen/NavigationScreen";
import ResourcesScreen from "./resources-screen/ResourcesScreen";
import ScrollScreen from "./scroll-screen/ScrollScreen";

const MainStack = createStackNavigator({
  Home: {
    screen: MainScreen, navigationOptions: {
      header: null,
    },
  },
  Resources: {
    screen: ResourcesScreen, navigationOptions: {
      header: null,
    },
  },
  Navigation: {
    screen: NavigationScreen, navigationOptions: {
      header: null,
    },
  },
  ListOfComponents: {
    screen: ListScreen, navigationOptions: {
      header: null,
    },
  },
  ScrollViewComponent: {
    screen: ScrollScreen, navigationOptions: {
      header: null,
    },
  },
  GridViewComponent: {
    screen: GridScreen, navigationOptions: {
      header: null,
    },
  },
  CameraComponent: {
    screen: CameraScreen, navigationOptions: {
      header: null,
    },
  },
  ImagePickerComponent: {
    screen: ImagePickerScreen, navigationOptions: {
      header: null,
    },
  },
  MapScreen: {
    screen: MapScreen, navigationOptions: {
      header: null,
    },
  },
});

const RootStack = createStackNavigator(
  {
    Main: {
      screen: MainStack,
    },
    NavigationModalScreen: {
      screen: NavigationScreen,
    },
  },
  {
    mode: "modal",
    headerMode: "none",
  },
);
export default RootStack;
