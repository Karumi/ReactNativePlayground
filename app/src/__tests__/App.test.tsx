import * as React from "react";
import * as renderer from "react-test-renderer";

import App from "../App";

it("renders correctly with defaults", () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
});
