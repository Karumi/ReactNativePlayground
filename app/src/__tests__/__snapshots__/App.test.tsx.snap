// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Open up App.js to start working on your app usign Typescript and Jest!
  </Text>
</View>
`;
